packages/flutter_tools/lib/src/web/chrome.dart

```diff
@@ -125,9 +125,9 @@ class ChromeLauncher {
     }
 
     final String chromeExecutable = findChromeExecutable(_platform, _fileSystem);
-    userDataDir ??= _fileSystem.systemTempDirectory.createTempSync('flutter_tools_chrome_device.');
 
-    if (cacheDir != null) {
+    if (userDataDir == null && cacheDir != null) {
```

Comment:
You also need something similar below on L163

Reply:
@jonahwilliams, do you think we shoud use the following Chrome flags only when the user doesn't pass a value to the `--web-user-data-dir` run flag? Or should we always use these flags in Chrome?

--disable-background-timer-throttling

--disable-extensions
--disable-popup-blocking
--bwsi
--no-first-run
--no-default-browser-check
--disable-default-apps
--disable-translate

Reply:
hmm, good question.

I would still provide `--disable-background-timer-throttling`, `--no-default-browser-check`. The others are somewhat questionable. If someone is using the user-data-dir to have access to their profile, then passing bwsi is counter productive

Reply:
... which is browser without sign in